src/Service/ConnectionHandler.cpp

```diff
@@ -687,20 +687,34 @@ void ConnectionHandler::updateStats(const Coordination::ZooKeeperResponsePtr & r
     if (response->xid != Coordination::WATCH_XID && response->getOpNum() != Coordination::OpNum::Heartbeat
         && response->getOpNum() != Coordination::OpNum::SetWatches && response->getOpNum() != Coordination::OpNum::Close)
     {
-        Int64 elapsed = Poco::Timestamp().epochMicroseconds() / 1000 - response->request_created_time_ms;
+        auto current_time = getCurrentTimeMilliseconds();
+        Int64 elapsed = current_time - response->request_created_time_ms;
+        if (elapsed < 0)
```

Comment:
Why elapsed can be -1? Caused by different clock? If it is, we must use `stedy_lock` in all places.

Reply:
Maybe different clock, so I use `getCurrentTimeMilliseconds` which is `steady_clock` for request in all places.

Reply:
Maybe caused by clock diff in different server. When leader append_entries the `create_time` is attached to the log and when the follower take the log the time in server is used. If there is a clock diff in the servers `-1` may raise.

Reply:
Let's add a coment to help reader understand the code.

Reply:
> Maybe caused by clock diff in different servers. When leader append_entries the `create_time` is attached to the log and when the follower takes the log the timestamp in leader is used. If there is a clock diff in the servers `-1` may raise.

When followers takes the log from leader,  them won't updateLatency for the log.We used to use `system_clock` to measure request time cost.